build/lib/android/snapshot.js

```diff
@@ -65,26 +60,20 @@ async function generateHeader() {
 	const blobs = {};
 
 	// Load snapshots for each architecture
-	for (const target of TARGETS) {
+	await Promise.all(TARGETS.map(async target => {
 		const V8_LIB_TARGET_DIR = path.resolve(V8_LIB_DIR, target);
 		const BLOB_PATH = path.join(V8_LIB_TARGET_DIR, 'blob.bin');
 
-		if (fs.existsSync(BLOB_PATH)) {
-			blobs[target] = Buffer.from(fs.readFileSync(BLOB_PATH, 'binary'), 'binary');
+		if (await fs.exists(BLOB_PATH)) {
+			blobs[target] = Buffer.from(await fs.readFile(BLOB_PATH, 'binary'), 'binary');
 		}
-	}
+	}));
 
-	return new Promise(async (resolve, reject) => {
-
-		// Generate 'V8Snapshots.h' from template
-		ejs.renderFile(path.join(__dirname, 'V8Snapshots.h.ejs'), blobs, {}, (error, output) => {
-			if (error) {
-				return reject(error);
-			}
-			fs.writeFileSync(path.join(ANDROID_DIR, 'runtime', 'v8', 'src', 'native', 'V8Snapshots.h'), output);
-			resolve();
-		});
-	});
+	console.log(`Generating V8Snapshots.h for ${Object.keys(blobs).join(', ')}...`);
+
+	// Generate 'V8Snapshots.h' from template
+	const output = await promisify(ejs.renderFile)(path.join(__dirname, 'V8Snapshots.h.ejs'), blobs, {});
+	return fs.writeFile(path.join(ANDROID_DIR, 'runtime', 'v8', 'src', 'native', 'V8Snapshots.h'), output);
```

Comment:
No need to address this, but just an FYI since I used to do the same thing. You don't need to worry about using '/' as the file separators in paths passed in to `path.join()` it does the right thing on Windows under the hood. I used to break each segment up live this thinking Node would be too stupid to apply the right separator otherwise, but it's not - it silently converts. And then you can avoid all the extra quoting/commas.